db/main/migrate/20180823000000_add_preferences.rb

```diff
@@ -0,0 +1,16 @@
+class AddPreferences < ActiveRecord::Migration[4.2]
+  def change
+    change_table :users do |t|
+      t.column :preferences, json_type, default: {}, null: false
```

Comment:
AFAIK adding a not-null default column is not a safe operation. I believe we need to make it nullable. See [Safe Operations For High Volume PostgreSQL](https://www.braintreepayments.com/blog/safe-operations-for-high-volume-postgresql/)

Side note: [This is going to be fixed in postgres 11](https://www.depesz.com/2018/04/04/waiting-for-postgresql-11-fast-alter-table-add-column-with-a-non-null-default/)!

Reply:
Oh no! It also seems setting a default in the same command as adding the column is not safe for big tables either. Hmm.

Reply:
Will split this into two commands and remove the NOT NULL. Thanks @igorwwwwwwwwwwwwwwwwwwww 👍